lib/erb.rb

```diff
@@ -548,7 +549,7 @@ class Buffer # :nodoc:
       def initialize(compiler, enc=nil)
         @compiler = compiler
         @line = []
-        @script = enc ? "#coding:#{enc}\n" : ""
+        @script = enc ? "#coding:#{enc}\n".dup : String.new
```

Comment:
@bquorning Yup, `dup` is not optimal way. So I used `String.new("#coding:#{enc}\n")`